Core/include/Acts/Propagator/RiddersPropagator.ipp

```diff
@@ -17,42 +17,31 @@ auto Acts::RiddersPropagator<propagator_t>::propagate(
       action_list_t_result_t<CurvilinearTrackParameters,
                              typename propagator_options_t::action_list_type>>;
 
+  auto startWithoutCov = start;
+  startWithoutCov.covariance() = std::nullopt;
+
   // Propagate the nominal parameters
-  auto nominalRet = m_propagator.propagate(start, options);
-  if (not nominalRet.ok()) {
-    return ThisResult::failure(nominalRet.error());
+  auto result = m_propagator.propagate(startWithoutCov, options);
+  if (not result.ok()) {
+    return ThisResult::failure(result.error());
   }
-
   // Extract results from the nominal propagation
-  auto nominalResult = std::move(nominalRet).value();
-  const BoundVector& nominalParameters =
-      nominalResult.endParameters->parameters();
-  // Use the curvilinear surface of the propagated parameters as target
-  const Surface& surface = nominalResult.endParameters->referenceSurface();
+  auto nominalResult = result.value();
+  assert(nominalResult.endParameters);
+  const auto& nominalFinalParameters = *nominalResult.endParameters;
 
-  // Steps for estimating derivatives
-  std::vector<double> deviations = {-4e-4, -2e-4, 2e-4, 4e-4};
-
-  // Allow larger distances for the oscillation
-  propagator_options_t opts = options;
-  opts.pathLimit *= 2.;
-
-  // Derivations of each parameter around the nominal parameters
-  std::array<std::vector<BoundVector>, eBoundSize> derivatives;
+  auto jacobian = wiggleAndCalculateJacobian(options, start, nominalResult);
```

Comment:
Generally, `auto` in combination with Eigen types has led to hard to debug bugs in the past (non evaluated expression templates etc), so ideally this would have an explicit type.